ecs-init/apparmor/apparmor_test.go

```diff
@@ -67,37 +70,90 @@ func TestLoadDefaultProfile(t *testing.T) {
 			loadError:        errors.New("mock load error"),
 			expectedError:    errors.New("mock load error"),
 		},
+		{
+			name:        "MissingTunablesGlobal",
+			profileName: "testProfile.txt",
+			statErrors: map[string]error{
+				"tunables/global": os.ErrNotExist,
+			},
+			profileContent: []string{"@{PROC}=/proc/", "#include <abstractions/base>"},
+		},
+		{
+			name:        "MissingAbstractionsBase",
+			profileName: "testProfile.txt",
+			statErrors: map[string]error{
+				"abstractions/base": os.ErrNotExist,
+			},
+			profileContent: []string{"#include <tunables/global>"},
+		},
+		{
+			name:        "MissingIncludes",
+			profileName: "testProfile.txt",
+			statErrors: map[string]error{
+				"tunables/global":   os.ErrNotExist,
+				"abstractions/base": os.ErrNotExist,
+			},
+			profileContent: []string{"@{PROC}=/proc/"},
+		},
+		{
+			name:        "StatError",
+			profileName: "testProfile.txt",
+			statErrors: map[string]error{
+				"tunables/global": os.ErrPermission,
+			},
+			expectedError: os.ErrPermission,
+		},
 	}
 	defer func() {
 		isProfileLoaded = aaprofile.IsLoaded
 		loadPath = aaparser.LoadProfile
 		createFile = os.Create
+		statFile = os.Stat
 	}()
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			if !apparmor.HostSupports() {
```

Comment:
Just curious, could you provide a bit more context as to why we're removing this check? (sorry not quite clear to me)

Reply:
`LoadDefaultProfile` reads/writes files in `/etc/apparmor.d`, which only exists from the get-go if the OS ships with AppArmor. The check would normally protect against accessing a directory that doesn't exist, but since we stub out the filesystem functions in the tests, the tests will never fail if the OS doesn't have AppArmor.